shared/templates/file_groupowner/ansible.template

```diff
@@ -4,6 +4,27 @@
 # complexity = low
 # disruption = low
 
+{{%- if GROUP_REPRESENTED_WITH_GID %}}
+- name: Set the newgroup variable if 0 found
+  set_fact:
+    newgroup: "{{{ GID_OR_NAME }}}"
+{{%- else %}}
+{{%- set GROUPS=GID_OR_NAME.split("|") %}}
+{{%- for grp in GROUPS %}}
+- name: Check that the {{{ grp }}} group is defined
+  getent:
+    database: group
+    key: {{{ grp }}}
+  ignore_errors: yes
+  when: newgroup is undefined
+
+- name: Set the newgroup variable if {{{ grp }}} found
+  set_fact: 
+    newgroup: "{{{ grp }}}"
```

Comment:
The reason of CI test failing seems to be somewhere around here. The fails happen in tests that apply Ansible Playbook for the profile. This Ansible Playbook contains Ansible Tasks for all rules in the given profile. Usually, there is multiple rules that use the file_groupowner template. The `newgroup` fact is set in all of these tasks. But, each time possibly to a different value. The tasks for different rules conflict with each other. Here you expect that the `newgroup` fact variable isn't defined but it has been defined by other rule. Please take a look. I think the problem could be solved by prefixing the name of the fact variable with the rule_id.

Reply:
Prefixed fact variable with rule_id